pkg/cmd/deployment/create/create.go

```diff
@@ -0,0 +1,191 @@
+package create
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/v2/api"
+	"github.com/cli/cli/v2/internal/ghrepo"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/spf13/cobra"
+)
+
+type createOptions struct {
+	BaseRepo   func() (ghrepo.Interface, error)
+	Branch     func() (string, error)
+	HttpClient func() (*http.Client, error)
+	IO         *iostreams.IOStreams
+
+	Ref         string
+	Environment string
+	Status      string
+}
+
+type deployment struct {
+	ID int `json:"id"`
+}
+
+func NewCmdCreate(f *cmdutil.Factory, runF func(*createOptions) error) *cobra.Command {
+	opts := createOptions{
+		HttpClient: f.HttpClient,
+		IO:         f.IOStreams,
+		Branch:     f.Branch,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create a new deployment",
+		Long: heredoc.Docf(`
+			Create a new deployment.
+
+			This will dispatch a %[1]sdeployment%[1]s event that external services can listen for and act on.
+		`, "`"),
+		Example: heredoc.Doc(`
+			# Create a deployment on the current branch
+			$ gh deployment create
+
+			# Create a deployment at a specified ref
+			$ gh deployment create --ref main
+
+			# Create a deployment for a specific environment
+			$ gh deployment create --environment test
+
+			# Create a deployment with an initial status
+			$ gh deployment create --status in_progress
+		`),
+		Args: cobra.NoArgs,
+		RunE: func(c *cobra.Command, args []string) error {
+			opts.BaseRepo = f.BaseRepo
+
+			// Validate the status flag
+			if opts.Status != "" {
+				switch opts.Status {
+				case "queued", "in_progress", "inactive", "in_review", "approved", "rejected", "pending", "error", "failure", "success":
+				default:
+					return fmt.Errorf("invalid status: %s", opts.Status)
+				}
+			}
+
+			if runF != nil {
+				return runF(&opts)
+			}
+			return createRun(&opts)
+		},
+	}
+
+	cmd.Flags().StringVarP(&opts.Ref, "ref", "r", "", "The `branch`, `tag` or `SHA` to deploy (default [current branch])")
```

Comment:
**Question**

I don't think it needs to be as part of this PR but would you agree that it might make sense to have flags for:
 * `task`
 * `auto_merge`
 * `required_contexts`
 * `paylod`
 * `description`
 * `transient_environment`
 * `production_environment`

As documented https://docs.github.com/en/rest/deployments/deployments?apiVersion=2022-11-28#create-a-deployment

Reply:
Possibly? In practice I'm not certain I'd make use of them.

To add context to my thinking on what has been included, my intent with this change is to be able to recreate what is achieved [in GitHub Actions when `environment` is defined for a job](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment#using-an-environment) in other CI providers such as CircleCI using the `gh` CLI. In Actions only the environment name and URL can be specified[^1], though likely some other fields are automatically populated?

[^1]: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idenvironment